api/src/main/java/org/pmiops/workbench/leonardo/LeonardoApiClientImpl.java

```diff
@@ -634,6 +646,44 @@ private Map<String, String> getBaseEnvironmentVariables(DbWorkspace workspace) {
 
     customEnvironmentVariables.putAll(buildCdrEnvVars(workspace.getCdrVersion()));
 
+    // if enableGkeApp is true
+    // Add variable with CROMWELL SERVER information to enable user to use cromwell as APP in
+    // jupyter
+    boolean flagEnabled = workbenchConfigProvider.get().featureFlags.enableGkeApp;
+    customEnvironmentVariables.put(CROMWELL_ENABLED, String.valueOf(flagEnabled));
+    if (flagEnabled) {
+      customEnvironmentVariables.put(CROMWELL_SERVER, getCromwellServer(workspace));
+    }
+
     return customEnvironmentVariables;
   }
+
+  private String getCromwellServer(DbWorkspace dbWorkspace) {
+    // Store the appName in DATABASE so that jupyter and cromwell server can use the same server
+    // information
+    return leonardoApiClientFactory.newApiClient().getBasePath()
+        + "v1/apps/"
+        + dbWorkspace.getGoogleProject()
+        + "/"
+        + getCromwellAppName(dbWorkspace);
+  }
+
+  private String getCromwellAppName(DbWorkspace dbWorkspace) {
```

Comment:
this is named "Cromwell" but it only makes calls to generically-named methods.  We should choose one or the other: either
* `getCromwellAppNameFromDb()` and `saveCromwellAppName()` or
* `getAppNameFromDb()` and `saveAppName()` with an appType parameter.

I don't see a real need to special-case Cromwell so I prefer generic with a parameter.